cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaStatusTest.java

```diff
@@ -108,6 +108,36 @@ public Kafka getKafkaCrd() throws ParseException {
                 .build();
     }
 
+    @Test
+    public void testStatusEntityOperatorReadiness() throws ParseException {
+        Kafka kafka = getKafkaCrd();
+        ResourceOperatorSupplier supplier = ResourceUtils.supplierWithMocks(false);
+
+        // Mock the Kafka Operator
+        CrdOperator mockKafkaOps = supplier.kafkaOperator;
+
+        when(mockKafkaOps.getAsync(eq(namespace), eq(clusterName))).thenReturn(Future.succeededFuture(getKafkaCrd()));
+
+        ArgumentCaptor<Kafka> kafkaCaptor = ArgumentCaptor.forClass(Kafka.class);
+        when(mockKafkaOps.updateStatusAsync(kafkaCaptor.capture())).thenReturn(Future.succeededFuture());
+
+        MockWorkingKafkaAssemblyOperator kao = new MockWorkingKafkaAssemblyOperator(vertx, new PlatformFeaturesAvailability(false, kubernetesVersion),
+                certManager,
+                supplier,
+                config);
+
```

Comment:
This test doesn't currently pass, even when I use the KafkaAssemblyOperator reconcile method. If I use the KafkaAssemblyOperator.reconcile(), shouldn't the returning future of createOrUpdaye() fail since there isn't an entityOperator is defined in the getKafkaCrd() method?

Reply:
I suspect you might need to write your own mock. If you check what is `MockWorkingKafkaAssemblyOperator` doing, you will probably find out that it never calls your method and therefore it never fails. The mocks here are intentionally written to simulate only certain situations and you might need to write your own mock to test exactly what you want.

i think you should also consider what do you want to test. The tests in this class basically simulate certain return value from the `reconcile()` method. The method `entityOperatorReady` is a level below. I wonder if more useful test would be to mock how it is checking the readiness and what does it return based on different pod states.

Reply:
Thanks for the advice, with it I was able to get a basic mock test working. Its hacky but I am getting a better hang of it. I'll build upon what I have learned so far to test the behaviour with different pods states in the next commit.